lib/updates/market.ts

```diff
@@ -0,0 +1,167 @@
+
+  const underlyingDecimals = await erc20Contract(
+    configs[token].underlyingAddress,
+    token,
+  ).decimals();
+  const targetUpdate: TargetUpdate = {
+    newTarget,
+    timestamp: currentBlock.timestamp,
+  };
+  const graphQLClient = createClient({
+    url: configs[token].paprSubgraph,
+  });
+  const res = await graphQLClient
+    .query<TargetUpdatesByControllerQuery>(TargetUpdatesByControllerDocument, {
+      controller: configs[token].controllerAddress,
+    })
+    .toPromise();
+
+  if (res.error) {
+    throw new Error(res.error.message);
+  }
+
+  const targetValues = targets(
+    res.data?.targetUpdates || [],
+    targetUpdate,
+    underlyingDecimals,
+  );
+  const l = targetValues.length;
+
+  const cur = targetValues[l - 1];
+  const prev = targetValues[l - 2];
+  const change = percentChange(prev.value, cur.value);
+  const apr = (change / (cur.time - prev.time)) * SECONDS_IN_A_YEAR;
+
+  // find the target update whose time is closest to an hour ago
+  const hourAgo = currentBlock.timestamp - 3600;
+  const targetHourAgo = targetValues.reduce((prev, curr) => {
+    return Math.abs(curr.time - hourAgo) < Math.abs(prev.time - hourAgo)
+      ? curr
+      : prev;
+  });
+  const targetPercentChange = percentChange(targetHourAgo.value, cur.value);
+  const direction = targetPercentChange > 0 ? '+' : '-';
+  return {
+    currentTarget: cur.value,
+    apr: formatPercent(apr),
+    targetPercentChange: `${direction}${formatPercent(targetPercentChange)}`,
+    targetHourAgo,
+  };
+}
+
+export async function getUniswapPoolInfo(token: SupportedToken) {
+  const provider = makeProvider(configs[token].jsonRpcProvider, token);
+  const currentBlock = await provider.getBlock('latest');
+
+  const graphQLClient = createClient({
+    url: configs[token].uniswapSubgraph,
+  });
+
+  const currentPoolRes = await graphQLClient
+    .query<PoolByIdQuery>(PoolByIdDocument, {
+      id: configs[token].uniswapPoolAddress,
+    })
+    .toPromise();
+  const pool = currentPoolRes.data?.pool;
+  if (currentPoolRes.error || !pool) {
+    throw new Error(currentPoolRes?.error?.message);
+  }
+  const baseCurrency =
+    getAddress(configs[token].underlyingAddress) == getAddress(pool.token0.id)
+      ? (pool.token1 as Token)
+      : (pool.token0 as Token);
+  const quoteCurrency =
+    getAddress(configs[token].underlyingAddress) == getAddress(pool.token0.id)
+      ? (pool.token0 as Token)
+      : (pool.token1 as Token);
+
+  const volumeUSD = currentPoolRes.data?.pool?.volumeUSD || 0;
+
+  const poolYesterdayRes = await graphQLClient
+    .query<PoolByIdByBlockQuery>(PoolByIdByBlockDocument, {
+      id: configs[token].uniswapPoolAddress,
+      blockHeight: currentBlock.number - BLOCKS_IN_A_DAY,
+    })
+    .toPromise();
+
+  if (poolYesterdayRes.error) {
+    throw new Error(poolYesterdayRes.error.message);
+  }
+  const volumeUSDYesterday = poolYesterdayRes.data?.pool?.volumeUSD || 0;
```

Comment:
@cnasc did we find that this is not accurate and we had to sum the hourly?

Reply:
We looked at summing the hourly data, but the fields came up as 0. My understanding is that this method is correct